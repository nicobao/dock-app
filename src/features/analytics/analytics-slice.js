import analytics from '@react-native-firebase/analytics';
import {captureException} from '@sentry/react-native';

export function logAnalyticsEvent(eventName, params = {}) {
  if (typeof eventName === 'string' && eventName.length > 0) {
    analytics()
      .logEvent(eventName, params)
      .catch(err => {
        captureException(err);
      });
  }
}
export const ANALYTICS_EVENT = {
  TOKENS: {
    SEND_TOKEN: 'token_send',
    RECEIVE_TOKEN: 'token_receive',
    BUY_TOKEN: 'token_buy',
  },
  WALLET: {
    BACKUP: 'wallet_backup',
    IMPORT: 'wallet_import',
  },
  ACCOUNT: {
    BACKUP: 'account_backup',
    IMPORT: 'account_import',
    EXPORT: 'account_export',
  },
  CREDENTIALS: {
    IMPORT: 'credentials_import',
    PRESENT: 'credentials_present',
  },
  SETTINGS: {
    TOGGLE_FEATURE_FLAG: 'settings_toggle_feature_flag',
    SWITCH_NETWORK: 'settings_switch_network',
  },
  FAILURES: 'failure',
  DID: {
    DID_CREATED: 'didCreated',
    DID_UPDATED: 'didUpdated',
    DID_IMPORTED: 'didUpdated',
    DID_SHARED: 'didShared',
    DID_EXPORTED: 'didExported',
  },
};
