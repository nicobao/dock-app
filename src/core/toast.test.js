import {shallow} from 'enzyme';
import {
  getErrorMessageFromErrorObject,
  renderToastElement,
  setToast,
  typeMap,
  withErrorToast,
} from './toast';
import {translate} from '../locales';
import analytics from '@react-native-firebase/analytics';

describe('Toast', () => {
  it('can get error message from error object', () => {
    const errorObject = new Error('Something went wrong');

    expect(getErrorMessageFromErrorObject(errorObject)).toBe(
      'Something went wrong',
    );

    expect(getErrorMessageFromErrorObject('Oops')).toBe('Oops');
    expect(
      getErrorMessageFromErrorObject(
        '[AssertionError: account already exists]',
      ),
    ).toBe('account already exists');

    expect(getErrorMessageFromErrorObject()).toBe(
      translate('global.unexpected_error'),
    );
  });
  describe('Error toast', () => {
    it('does wrapper error log errors', async () => {
      const errFunction = () => {
        throw new Error('Test Error');
      };
      const func = withErrorToast(errFunction);

      await expect(func('Sample argument')).rejects.toThrowError('Test Error');
      expect(analytics().logEvent).toBeCalledWith('failure', {
        0: 'Sample argument',
        message: 'Test Error',
      });
    });
  });

  describe('toast element', () => {
    beforeAll(() => {
      setToast({
        closeAll: () => {},
      });
    });

    it('should render without errors', () => {
      const message = 'Some message';
      const wrapper = shallow(
        renderToastElement({typeProps: typeMap.success, message}),
      );

      expect(wrapper.dive()).toMatchSnapshot();
    });

    it('should handle non string messages without errors', () => {
      const message = {test: true};
      const wrapper = shallow(
        renderToastElement({typeProps: typeMap.success, message}),
      );

      expect(wrapper.dive()).toMatchSnapshot();
    });

    it('should handle undefined messages without errors', () => {
      const message = 'Some message';
      const wrapper = shallow(
        renderToastElement({typeProps: typeMap.success, message}),
      );

      expect(wrapper.dive()).toMatchSnapshot();
    });
  });
});
