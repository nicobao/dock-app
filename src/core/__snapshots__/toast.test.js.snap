// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast toast element should handle non string messages without errors 1`] = `
<View
  _pressed={
    Object {
      "opacity": 0.2,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Memo(ForwardRef(Stack))
    bg="#27272A"
    direction="row"
    mb={5}
    px={6}
    py={4}
    rounded="md"
  >
    <SvgFile />
    <Memo(ForwardRef(Text))
      fontSize={14}
      fontWeight={600}
      ml={2}
      style={
        Object {
          "color": "#A1A1AA",
        }
      }
    >
      {"test":true}
    </Memo(ForwardRef(Text))>
  </Memo(ForwardRef(Stack))>
  <PressabilityDebugView
    color="red"
  />
</View>
`;

exports[`Toast toast element should handle undefined messages without errors 1`] = `
<View
  _pressed={
    Object {
      "opacity": 0.2,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Memo(ForwardRef(Stack))
    bg="#27272A"
    direction="row"
    mb={5}
    px={6}
    py={4}
    rounded="md"
  >
    <SvgFile />
    <Memo(ForwardRef(Text))
      fontSize={14}
      fontWeight={600}
      ml={2}
      style={
        Object {
          "color": "#A1A1AA",
        }
      }
    >
      Some message
    </Memo(ForwardRef(Text))>
  </Memo(ForwardRef(Stack))>
  <PressabilityDebugView
    color="red"
  />
</View>
`;

exports[`Toast toast element should render without errors 1`] = `
<View
  _pressed={
    Object {
      "opacity": 0.2,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Memo(ForwardRef(Stack))
    bg="#27272A"
    direction="row"
    mb={5}
    px={6}
    py={4}
    rounded="md"
  >
    <SvgFile />
    <Memo(ForwardRef(Text))
      fontSize={14}
      fontWeight={600}
      ml={2}
      style={
        Object {
          "color": "#A1A1AA",
        }
      }
    >
      Some message
    </Memo(ForwardRef(Text))>
  </Memo(ForwardRef(Stack))>
  <PressabilityDebugView
    color="red"
  />
</View>
`;
